### PR TITLE
[Feature] Add adjust_hue

### PR DESCRIPTION
**Pull request by teamwong111**
## Motivation

Impl adjust hue.

## Modification

```
mmcv/image/__init__.py
mmcv/image/photometric.py
tests/test_image/test_photometric.py
```

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by xiaojieli0903** [comment]
hi @zhouzaida  when will this PR be approved?

**Comment by zhouzaida** [comment]
> hi @zhouzaida when will this PR be approved?

Hi @xiaojieli0903 , sorry for the late reply. We will review this PR ASAP.

File: mmcv/image/photometric.py [hunk 1]
```diff
@@ -426,3 +426,39 @@ def clahe(img, clip_limit=40.0, tile_grid_size=(8, 8)):
...
     clahe = cv2.createCLAHE(clip_limit, tile_grid_size)
     return clahe.apply(np.array(img, dtype=np.uint8))
+
+
+def adjust_hue(img: np.ndarray, hue_factor: float) -> np.ndarray:
+    """Adjust hue of an image. The image hue is adjusted by converting the
+    image to HSV and cyclically shifting the intensities in the hue channel
+    (H). The image is then converted back to original image mode. `hue_factor`
+    is the amount of shift in H channel and must be in the interval `[-0.5,
+    0.5]`.
```
Comment (zhouzaida) [inline mmcv/image/photometric.py:436]:
```suggestion
    """Adjust hue of an image. 
    
    The image hue is adjusted by converting the image to HSV and cyclically
    shifting the intensities in the hue channel (H). The image is then converted
    back to original image mode.
    
    `hue_factor` is the amount of shift in H channel and must be in the interval `[-0.5,
    0.5]`.
```

File: mmcv/image/photometric.py [hunk 2]
```diff
@@ -426,3 +426,42 @@ def clahe(img, clip_limit=40.0, tile_grid_size=(8, 8)):
...
+
+def adjust_hue(img: np.ndarray, hue_factor: float) -> np.ndarray:
+    """Adjust hue of an image. 
+    
+    The image hue is adjusted by converting the image to HSV and cyclically
+    shifting the intensities in the hue channel (H). The image is then converted
+    back to original image mode.
+    
+    `hue_factor` is the amount of shift in H channel and must be in the interval `[-0.5,
+    0.5]`.
```
Comment (zhouzaida) [inline mmcv/image/photometric.py:439]:
```suggestion
    """Adjust hue of an image.

    The image hue is adjusted by converting the image to HSV and cyclically
    shifting the intensities in the hue channel (H). The image is then
    converted back to original image mode.

    `hue_factor` is the amount of shift in H channel and must be in the
    interval `[-0.5, 0.5]`.
```

**Comment by xiaojie1017** [comment]
> > hi @zhouzaida when will this PR be approved?
> 
> Hi @xiaojieli0903 , sorry for the late reply. We will review this PR ASAP.

thanks!!

File: mmcv/image/photometric.py [hunk 3]
```diff
@@ -426,3 +426,42 @@ def clahe(img, clip_limit=40.0, tile_grid_size=(8, 8)):
...
+        img (ndarray): Image to be adjusted.
+        hue_factor (float):  How much to shift the hue channel. Should be in
+            [-0.5, 0.5]. 0.5 and -0.5 give complete reversal of hue channel in
+            HSV space in positive and negative direction respectively.
+            0 means no shift. Therefore, both -0.5 and 0.5 will give an image
+            with complementary colors while 0 gives the original image.
+
+    Returns:
+        ndarray: Hue adjusted image.
+    """
```
Comment (HAOCHENYE) [inline mmcv/image/photometric.py:451]: Should we add some copy right information, modified from [torchvision.transforms.functional.adjust_hue](https://github.com/pytorch/vision/blob/651b97e2d0ebb1594ba5a318a6df69cb5129588a/torchvision/transforms/functional.py#L864)

Comment (teamwong111) [reply mmcv/image/photometric.py]: Updated.

**Review by zhouzaida — APPROVED** [review]

**Pull request merged**